src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,365 @@
+---
+title: First Discovery Server
+layout: default
+category: Overview
+---
+
+## Launching ##
+
+The First Discovery Server can be launched as a [Kettle](https://github.com/amb26/kettle/blob/KETTLE-32/README.md) application by making use of [Kettle Configs](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#structure-of-a-kettle-config).
+
+There are two typical ways of launching a Kettle app, programmatically and from command line.
+
+(See: [Starting a Kettle application](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#starting-a-kettle-application))
+
+### Programmatically ###
+
+```javascript
+// require the kettle module
+var kettle = require("kettle");
+
+// load the config
+kettle.config.loadConfig({
+    // path to the config directory
+    configPath:"./src/config",
+
+    // name of the config to load, without the file extension
+    configName: "vagrant"
+});
+```
+
+### Command Line ####
+
+```bash
+# Call Kettle's init.js script with the
+# configPath and configName
+node node_modules/kettle/init.js <configPath> [<configName>]
+
+# or using an environment variable to specify
+# the configName
+NODE_EVN=<configName> node node_modules/kettle/init.js <configPath>
+```
+
+## Configuration ##
+
+The First Discovery Server can be configured via [Kettle Configs](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#structure-of-a-kettle-config). A set of these are provided with the server in the [config](https://github.com/GPII/first-discovery-server/tree/master/src/config) directory. An integrator can choose to make use of the provide configs directly or use them as the basis for creating new configs.
+
+The [`gpii.firstDiscovery.server.configurator`](./src/js/firstDiscoveryServer.js) grade defines a default schema for which the configuration is validated against. If the validation fails, the application will throw and error.
+
+For more information about how the Preferences Server's security layer is configured. Please see the [Client Credentials Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Client_Credentials_Grant) documentation.
+
+### environment.json ###
+
+Accepts environment variables for all of the critical configuration options that are needed for running the First Discovery Server and connecting to the Oauth2 security server.
+
+(see: [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json))
+
+<table>
+    <thead>
+        <tr>
+            <th>Option</th>
+            <th>Environment Variable</th>
+            <th>Description</th>
+            <th>Example</th>
+        <tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                <code>"port"</code>
+            </td>
+            <td>
+                <code>FIRST_DISCOVERY_SERVER_TCP_PORT</code>
+            </td>
+            <td>
+                The port number that the First Discovery Server is hosted at.
+            </td>
+            <td>
```

Comment:
Brainstorming ways to reconfigure this table: What would it look like if we stated the environment variable as part of the Description? i.e. remove the Env Var column and in the Description column, add a sentence something like "Sets the XXX environment variable" or whatever would be technically correct...